### PR TITLE
Deprecate TemporaryFile context manager and tmpfile test fixture

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR deprecates the `gwpy.testing.utils.TemporaryFile` context manager function and the `gwpy.testing.fixtures.tmpfile` test fixture. These are basically no longer required now that we can rely properly on the [`tmp_path`](https://docs.pytest.org/en/stable/tmpdir.html?) fixture from pytest.

### CODE REVIEW

**Comment by eagoetz** [comment]
@duncanmmacleod we understand this is still a draft and that you may edit this further after #1362 gets merged. We'll come back to this with you once that is done.

**Comment by duncanmmacleod** [comment]
@gwpy/maintainers this is now ready for review.

**Review by pesummary — APPROVED** [review]
LGTM.

**Review by hoyc1 — APPROVED** [review]

**Pull request merged**
